Backend/app/rider-platform/rider-app/Main/src/Storage/Queries/Booking.hs

```diff
@@ -104,10 +104,18 @@ updateOtpCodeBookingId rbId otp = do
     ]
     [Se.Is BeamB.id (Se.Eq $ getId rbId)]
 
-findLatestByRiderIdAndStatus :: (MonadFlow m, CacheFlow m r, EsqDBFlow m r) => Id Person -> [BookingStatus] -> m (Maybe BookingStatus)
-findLatestByRiderIdAndStatus (Id riderId) bookingStatusList =
+findLatestByRiderId :: (MonadFlow m, CacheFlow m r, EsqDBFlow m r) => Id Person -> m (Maybe BookingStatus)
+findLatestByRiderId (Id riderId) =
   do
-    let options = [Se.And [Se.Is BeamB.riderId $ Se.Eq riderId, Se.Is BeamB.status $ Se.In bookingStatusList]]
+    let options =
+          [ Se.And
+              [ Se.Is BeamB.riderId $ Se.Eq riderId,
+                Se.Or
```

Comment:
In case of older booking, isScheduled will nothing. Handle that too as False case.

Reply:
This is already handled as 2nd condition is having equality for `not (Just True)`. So `Nothing` will pass this `condition`